cmd/kubeadm/app/master/selfhosted_volumes.go

```diff
@@ -0,0 +1,294 @@
+									Path: path.Join("pki", kubeadmconstants.ServiceAccountPrivateKeyName),
+								},
+							},
+						},
+					},
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.FrontProxyCACertAndKeyBaseName,
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  "tls.crt",
+									Path: path.Join("pki", kubeadmconstants.FrontProxyCACertName),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func schedulerProjectedVolume() *v1.Volume {
+	return &v1.Volume{
+		Name: "k8s",
+		VolumeSource: v1.VolumeSource{
+			Projected: &v1.ProjectedVolumeSource{
+				Sources: []v1.VolumeProjection{
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.SchedulerKubeConfigFileName,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func controllerManagerProjectedVolume() *v1.Volume {
+	return &v1.Volume{
+		Name: "k8s",
+		VolumeSource: v1.VolumeSource{
+			Projected: &v1.ProjectedVolumeSource{
+				Sources: []v1.VolumeProjection{
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.ControllerManagerKubeConfigFileName,
+							},
+						},
+					},
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.CACertAndKeyBaseName,
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  "tls.crt",
+									Path: path.Join("pki", kubeadmconstants.CACertName),
+								},
+								{
+									Key:  "tls.key",
+									Path: path.Join("pki", kubeadmconstants.CAKeyName),
+								},
+							},
+						},
+					},
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.ServiceAccountKeyBaseName,
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  "tls.key",
+									Path: path.Join("pki", kubeadmconstants.ServiceAccountPrivateKeyName),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func createTLSSecrets(client *clientset.Clientset) error {
+	tlsPairs := []string{
+		kubeadmconstants.CACertAndKeyBaseName,
+		kubeadmconstants.CACertName,
+		kubeadmconstants.CAKeyName,
+		kubeadmconstants.APIServerCertAndKeyBaseName,
+		kubeadmconstants.APIServerCertName,
+		kubeadmconstants.APIServerKeyName,
+		kubeadmconstants.APIServerKubeletClientCertAndKeyBaseName,
+		kubeadmconstants.APIServerKubeletClientCertName,
+		kubeadmconstants.APIServerKubeletClientKeyName,
+		kubeadmconstants.ServiceAccountKeyBaseName,
+		kubeadmconstants.ServiceAccountPublicKeyName,
+		kubeadmconstants.ServiceAccountPrivateKeyName,
+		kubeadmconstants.FrontProxyCACertAndKeyBaseName,
+		kubeadmconstants.FrontProxyCACertName,
+		kubeadmconstants.FrontProxyCAKeyName,
+	}
+	for i := 0; i < len(tlsPairs); i = i + 3 {
```

Comment:
You can make a small, internal struct for storing this information and do `for _, tlsPair := range ` loop, that'd be better IMO.